base/linalg/arpack.jl

```diff
@@ -228,7 +228,7 @@ for (T, saupd_name, seupd_name, naupd_name, neupd_name) in
                   sizeof(howmny), sizeof(bmat), sizeof(evtype))
         end
 
-        function saupd(ido, bmat, n, which, nev, TOL::Array{$T}, resid::Array{$T}, ncv, v::Array{$T}, ldv,
+        function saupd(ido, bmat, n, which, nev, TOL::Ref{$T}, resid::Array{$T}, ncv, v::Array{$T}, ldv,
```

Comment:
As mentioned earlier, I don't think `TOL` needs to be a `Ref`. It is only [an input parameter](http://www.caam.rice.edu/software/ARPACK/UG/node136.html#SECTION001210000000000000000), so I think we should revert to the `Array{$T}` for the 0.6 release and then change it to `T` when 0.6 has branched.